trick_source/java/src/trick/common/utils/vs/VSDouble.java

```diff
@@ -12,6 +12,23 @@ public VSDouble(double value) {
 
     @Override
     public void fromVariableServer(String string) {
+        // CTLaChance
+        // JUN 07 2017
+        /*
+        Converts the C representation of +- infinity and +- not-a-number to the Java representation
+        so that Double.parseDouble() can correctly assign those values to the variable value.
+        -NaN does not exist in Java and is therefore converted to NaN.
+        */
+        if(string.equals("inf")) {
```

Comment:
This code is replicated in `VSFloat.fromVariableServer`, and I have a special hatred in my heart for code duplication. I recommend lifting this logic into a standalone protected function in the super class.